src/test_application/VehicleBuilder.cpp

```diff
@@ -226,6 +229,10 @@ void VehicleBuilder::finalize_machines()
             }
         }
     }
+
+    VehicleData dataOut{std::move(*m_data)};
```

Comment:
Consider using std::exchange here

Reply:
I recall attempting that, but can't quite figure it out. VehicleData is move-only, m_data is an optional saying the move assignment operator is deleted.
```cpp
// error: use of deleted function ‘std::optional<testapp::VehicleData>& std::optional<testapp::VehicleData>::operator=(std::optional<testapp::VehicleData>&&)’
VehicleData dataOut{std::exchange(m_data, {}).value()};
```